lib/elixir/lib/task.ex

```diff
@@ -338,13 +341,17 @@ defmodule Task do
   end
 
   @doc """
-  Receives a group of tasks and a message and finds
+  Takes a list of tasks and a message and finds
   a task that matches the given message.
 
-  This function returns a tuple with the returned value
-  in case the message matches a task that exited with
-  success alongside the matching task. It returns `nil`
-  if no task was found. It exits if the task has failed.
+  If the message matches a task that finished normally,
+  this function returns a tuple with the task's
+  result and `Task` struct.
+
+  If the message matches a task that terminated before
+  sending its result back, this function will exit.
```

Comment:
So yeah, it appears I'm still not 100% sold on the idea that task errors should propagate to the caller. Sorry, @fishcakez! I was 99% of the way there, but then I found out `yield` had been changed to not always exit in Elixir 1.2.

So I guess that choice between the exiting `await` and a non-exiting `yield` confuses me when I'm thinking of the right way to use tasks. And it's especially confusing in the case of `find` (which is supposed to be used in a gen server) since killing a gen server is something you don't normally expect. Normally, when someone is talking about a gen server spawning tasks, they imply receiving results from those tasks as messages and not killing the whole server because of one of the tasks terminating.